py-polars/polars/exceptions.py

```diff
@@ -53,6 +54,9 @@ class StringCacheMismatchError(Exception):  # type: ignore[no-redef]
     class StructFieldNotFoundError(Exception):  # type: ignore[no-redef]
         """Exception raised when a specified schema field is not found."""
 
+    class PerformanceWarningCategoricalRemapping(Warning):  # type: ignore[no-redef]
```

Comment:
Ideally we'd have a `PerformanceWarning` class and some subclasses such as `CategoricalRemappingWarning`. I think trying to put it all in one name is a bit much (and IF we do that it should be `CategoricalRemappingPerformanceWarning`, imo).

Perhaps we can start with a more generic `PerformanceWarning` and add other warnings when we figure out subclassing? I actually have some pointers from the PyO3 people on how to do it but haven't gotten around to it..